rbd/admin/admin.go

```diff
@@ -49,3 +49,57 @@ func NewLevelSpec(pool, namespace, image string) LevelSpec {
 func NewRawLevelSpec(spec string) LevelSpec {
 	return LevelSpec{spec}
 }
+
+// ImageSpec values are used to identify RBD image wherever Ceph APIs
+// require an image_spec using image name and optional pool and namespace.
+type ImageSpec struct {
+	spec string
+}
+
+// ImageIDSpec values are used to identify RBD image wherever Ceph APIs
+// require an image_id_spec using image id and optional pool and namespace.
+type ImageIDSpec struct {
+	spec string
+}
+
+// parseImageSpec is used to construct a ImageSpec given a image and
+// optional namespace and pool names.
+func parseImageSpec(pool, namespace, image string) string {
```

Comment:
Isn't this the LevelSpec type that already exists?

Reply:
Nope, image[id]spec needs image name/ID with optional pool and namespace.

Hope my description was clear enough. 

Tasks are added per image.

Reply:
LevelSpec is also an image id with optional pool and namespace, see here: https://github.com/ceph/go-ceph/blob/e0e56486b0fb72bd4977f4b162ae4702fb78e1b9/rbd/admin/admin.go#L31-L43
So, where is the difference?

Reply:
As a side note: this is a string constructor, not a parser, I would say.

Reply:
Levelspec is more of required pool with optional ns and image.( mirror is enabled/ schedule added per [ns ed] pool/ image. 

While image[id]spec requires image with optional pool and ns.
(Task is added per image)

Actual ceph extraction code here.
https://github.com/ceph/ceph/blob/a67d1cf2a7a4031609a5d37baa01ffdfef80e993/src/pybind/mgr/rbd_support/task.py#L185


And formatter to output image_[id_]spec 

https://github.com/ceph/ceph/blob/a67d1cf2a7a4031609a5d37baa01ffdfef80e993/src/pybind/mgr/rbd_support/task.py#L222